src/main/scala/is/hail/asm4s/package.scala

```diff
@@ -176,6 +176,9 @@ package object asm4s {
 
   implicit def toCodeBoolean(c: Code[Boolean]): CodeBoolean = new CodeBoolean(c)
 
+  // on the JVM Booleans are represented as Ints
+  implicit def booleanToCodeInt(c: Code[Boolean]): CodeInt = new CodeInt(c.asInstanceOf[Code[Int]])
```

Comment:
I don't think I like this.  We should have `toInt` on `Code[Boolean]` analogous to what we have in Scala.  It can be implemented as a no-op.

Reply:
I went back and forth myself on this. I'll go with your suggestion but, in general, I think it's wrong to treat `Code[T]` as analogous to `T` because the difference between byte code and Scala code is a sufficiently large gap that when I'm working with `Code[T]` I think at the level of byte code. I don't think that's bad, I like that I can operate below the level of Scala.

Reply:
I'm confused.  This isn't about the level that you're working, it is implicit vs explicit as the user of `Code`, same as whether we have an implicit to explicit conversion to boolean in our Scala code.  The generated bytecode is the same.  I would have assumed you'd want it explicit, too.  You've come along way, Mr. Types.

 > I think it's wrong to treat Code[T] as analogous to T because the difference between byte code and Scala code is a sufficiently large gap that when I'm working with Code[T] I think at the level of byte code.

Well, when I write Scala I think at the bytecode level, too, but I don't think that's what you mean.  I think this is largely a failure of my design for `Code`, which was to do type-safe and type-enabled bytecode generation, so we could do things like `regionCode.loadBit(offCode, bitCode)`, etc.  Either `Code[T]` was badly designed for your purposes or it was the wrong abstraction altogether (and we should have something like `UntypedCode` or just use `asm` directly), I'm not 100% sure.  Having two incompatible purposes push on `Code` will only lead to a library that is confusing to its users and confused in its uses.

To be clear, my intention was for `Code[T]` to represent a sequence of bytecode which, when executed, leaves a `T` on the top of the stack.

Reply:
I guess what I mean is that when I'm thinking in Code-world, Int and Boolean are the same thing. Conversion instructions seem like noise. `Int` == `Boolean` == 32-bits of signed arithmetic.

In Scala-world, I don't think about byte code, I think of Boolean as the type with two inhabitants.